src/functional/ffront/type_info.py

```diff
@@ -171,6 +171,32 @@ def is_integral(symbol_type: ct.SymbolType) -> bool:
     ]
 
 
+def is_number(symbol_type: ct.SymbolType) -> bool:
+    """
+    Check if ``symbol_type`` is either intergral or float.
+
+    Examples:
+    ---------
+    >>> is_number(ct.ScalarType(kind=ct.ScalarKind.FLOAT64))
+    True
+    >>> is_number(ct.ScalarType(kind=ct.ScalarKind.INT32))
+    True
+    >>> is_number(ct.ScalarType(kind=ct.ScalarKind.BOOL))
+    False
+    >>> is_number(ct.FieldType(dims=[], dtype=ct.ScalarType(kind=ct.ScalarKind.INT)))
+    False
+    """
+    if not isinstance(symbol_type, ct.ScalarType):
+        return False
+    return symbol_type.kind in [
+        ct.ScalarKind.INT,
+        ct.ScalarKind.INT32,
+        ct.ScalarKind.INT64,
+        ct.ScalarKind.FLOAT32,
+        ct.ScalarKind.FLOAT64,
+    ]
```

Comment:
```suggestion
    if not isinstance(symbol_type, ct.ScalarType):
        return False
    return is_arithmetic(symbol_type)
```
To avoid code duplication.

Reply:
it is not exactly linear as `is_arithmetic`returns true for emtpy fields

Reply:
this is a fair point, I will add a TODO here and change it the way you suggested. I want to do these edits in a separate PR